server/auth_test.go

```diff
@@ -49,6 +49,23 @@ func TestUserClone(t *testing.T) {
 	}
 }
 
+func TestUserClonePermissionsNoLists(t *testing.T) {
+	user := &User{
+		Username:    "foo",
+		Password:    "bar",
+		Permissions: &Permissions{},
+	}
+
+	clone := user.clone()
+
+	if clone.Permissions.Publish != nil {
+		t.Fatalf("Expected Publish to be nil, got: %v", clone.Permissions.Publish)
+	}
+	if clone.Permissions.Subscribe != nil {
+		t.Fatalf("Expected Subscribe to be nil, got: %v", clone.Permissions.Subscribe)
+	}
```

Comment:
Can you add a test where Permissions is nil (not empty like in this test) and that the clone's Permissions is nil too?